openspoor/visualisations/spoorkaart.py

```diff
@@ -0,0 +1,222 @@
+        :param radius_column: A column noting the radius of the circles to plot (if markertype=='circle')
+        :param url_column: A column including an url that is displayed in the popup
+        """
+        super().__init__(df)
+        self.attrs['lat'] = lat_column
+        self.attrs['lon'] = lon_column
+        if 'geometry' in self.columns:
+            if isinstance(self.geometry[0], point.Point):
+                self[self.attrs['lat']] = self.geometry.apply(lambda d: d.y)
+                self[self.attrs['lon']] = self.geometry.apply(lambda d: d.x)
+            else:
+                NotImplementedError(f"Unimplemented geometry: {self.geometry[0]}")
+
+        self.color_column = color_column
+        if self.color_column is not None:
+            self[self.color_column] = pd.factorize(self[self.color_column])[0]
+
+        # TODO: Automatically do this by looping through args?
+        self.attrs['popup'] = popup
+        self.attrs['markertype'] = markertype
+        self.attrs['colors'] = colors
+        self.attrs['marker'] = marker_column
+        self.attrs['rotation'] = rotation_column
+        self.attrs['radius'] = radius_column
+        self.attrs['url_column'] = url_column
+
+    def add_to(self, folium_map):
+        for i, row in self.iterrows():
+            location = list(row[[self.attrs['lat'], self.attrs['lon']]].values)
+            if self.attrs['popup']:
+                popup_text = ''
+                for col in self.attrs['popup']:
+                    if col == self.attrs['url_column']:
+                        url = quote(row[col], safe='/:?=&') # Replaces characters unsuitable for URL's
+                        popup_text = popup_text + f"{col}: <a href={url}>Hyperlink</a><br>",
+                    else:
+                        popup_text = popup_text + f'{col}: {row[col]}<br>'
+            else:
+                popup_text = None
+
+            if self.attrs['rotation'] is not None:
+                rotation = int(row[self.attrs['rotation']])
+                marker = 'arrow-up'
+            else:
+                rotation = 0
+                if self.attrs['marker'] is not None:
+                    marker = row[self.attrs['marker']]
+                elif self.attrs['markertype'] is not None:
+                    marker = self.attrs['markertype']
+                else:
+                    marker = config['default_marker']
+
+            if self.color_column is not None:
+                colorset = ['purple', 'lightblue', 'darkgreen', 'blue', 'darkred', 'black',
+                            'pink', 'cadetblue', 'lightgray', 'lightred', 'green',
+                            'beige', 'darkblue', 'darkpurple', 'orange', 'lightgreen', 'red']
+                marker_color = colorset[int(row[self.color_column]) % len(colorset)]
+            elif self.attrs['colors'] is not None:
+                if isinstance(self.attrs['colors'], str):
```

Comment:
folium 0.2 has various colormap functions, and for folium 0.12 you can use libraries that do the same
see [folium quickstart](https://python-visualization.github.io/folium/quickstart.html#Styling-function)

Reply:
Markercolors seem to work only for a pre-defined list; it looks like the libraries (I found colorbrewer for example) only work for chloropleth maps.